src/render/glyph_atlas.js

```diff
@@ -20,22 +20,31 @@ export type GlyphPosition = {
     metrics: GlyphMetrics
 };
 
-export type GlyphPositions = { [string]: { [number]: GlyphPosition } }
+export type GlyphPositionData = {
+    glyphPositionMap: { [number]: GlyphPosition },
+    ascender: number,
+    descender: number
+};
+
+export type GlyphPositions = { [string]: GlyphPositionData };
 
 export default class GlyphAtlas {
     image: AlphaImage;
     positions: GlyphPositions;
 
-    constructor(stacks: { [string]: { [number]: ?StyleGlyph } }) {
+    constructor(stacks: {[string]: {glyphs: {[number]: ?StyleGlyph}, ascender: number, descender: number}}) {
```

Comment:
`{[number]: ?StyleGlyph}, ascender: number, descender: number}`

This could be condensed into a new type given that it is used in multiple places in the codebase.